scripts/tasks/codepen-examples-transform.js

```diff
@@ -0,0 +1,112 @@
+/*
+This Transform modifies Fabric website code examples into a format that will allow them to be rendered on Codepen, as part of the "Export to Codepen" feature.
+There are two types of supported example templates:
+1)
+[imports]
+[variable export named declaration with example code inside]
+and
+2)
+[imports]
+[class export named declaration with example code inside. Having multiple methods and keeping track of state is supported]
+Currently, examples which are scattered across multiple files are NOT supported.
+*/
+const babylon = require('babylon');
+const recast = require('recast');
+
+function parseRaw(code) {
+  const parse = source =>
+    babylon.parse(source, {
+      sourceType: 'module',
+      plugins: ['jsx', 'typescript', 'classProperties']
+    });
+  return recast.parse(code, { parser: { parse } }).program.body;
+}
+
+function transform(file, api) {
+  const parse = source =>
+    babylon.parse(source, {
+      sourceType: 'module',
+      plugins: ['jsx', 'typescript', 'classProperties']
+    });
+
+  const j = api.jscodeshift;
+
+  let source = j(recast.parse(file.source, { parser: { parse } }));
+
+  //remove react import
+  source = source
+    .find(j.ImportDeclaration, node => node.source.value == 'react')
+    .remove()
+    .toSource();
+  source = j(recast.parse(source, { parser: { parse } }));
+
+  //remove css imports
+  source = source
+    .find(j.ImportDeclaration, node => node.source.value.endsWith('css'))
+    .remove()
+    .toSource();
+  source = j(recast.parse(source, { parser: { parse } }));
+
+  // attach all imported components to the window
+  let attachedWindowString = 'let {\n';
+
+  let imports = source.find(j.ImportDeclaration);
+
+  let identifiers = [];
+  imports.forEach(p => {
+    p.node.specifiers.forEach(spec => {
+      identifiers.push(spec.local.loc.identifierName);
+    });
+  });
+
+  for (var i = 0; i < identifiers.length; i++) {
+    attachedWindowString += '\t' + identifiers[i] + ',\n';
+  }
+  attachedWindowString += '\tFabric } = window.Fabric;';
+
+  let parsedAttachedWindowString = parseRaw(attachedWindowString);
+
+  // remove the rest of the import declarations
+  source = source
+    .find(j.ImportDeclaration)
+    .remove()
+    .insertBefore(parsedAttachedWindowString)
+    .toSource();
+  source = j(recast.parse(source, { parser: { parse } }));
+
+  // remove exports and replace with variable or class declarations, whichever the original example used
+  let variableExportDeclarations = source.find(
+    j.ExportNamedDeclaration,
+    node => node.declaration.type == 'VariableDeclaration'
+  );
+  let classExportDeclarations = source.find(
+    j.ExportNamedDeclaration,
+    node => node.declaration.type == 'ClassDeclaration'
+  );
+  let exampleName;
+
+  // for examples with variable export declarations
+  if (variableExportDeclarations.__paths.length > 0) {
```

Comment:
Is there something a bit less "internal" from variableExportDeclarations that you can test for? Is this an array or an iterable?

Reply:
yup, size() seems to be a wrapper around __paths.length, which is a lot less sketchy than what i was trying to do